app/views/admin/base.py

```diff
@@ -82,6 +104,27 @@ def get_params(self):
             self.request,
             post_params={'xsrf_token': utils.strip})
 
+    def _enforce_admin_level(self, min_level):
+        if self.env.user_admin_permission is None:
+            raise django.core.exceptions.PermissionDenied
+        if self.env.user_admin_permission.compare_level_to(min_level) < 0:
+            raise django.core.exceptions.PermissionDenied
+
+    def enforce_moderator_admin_level(self):
```

Comment:
if you wanted to be super fancy, maybe you could turn these methods into annotations (like @RequireJobOrSuperuser, not that I've seen that name in a codebase or something)

Reply:
Oh, yeah, done.